README.md

```diff
@@ -76,6 +76,23 @@ steps:
             - /workdir/node_modules
 ```
 
+It also supports caching on specific JSON keys which can be specified following a `#` character using [jq syntax](https://jqlang.github.io/jq/manual/#object-identifier-index). This requires [jq](https://jqlang.github.io/jq/) to be installed on the build agent. You cannot use this in conjunction with Bash globbing.
```

Comment:
More for our internal understanding, is it worth clarifying

- The implementation matches on the first `.json#` substring, which would be silly to have elsewhere in your file path but is nonetheless possible on Unix systems
- A given entry cannot mix globbing and jq paths, but you can do `['**/LICENSE.md', 'package.json#.dependencies']`